src/main/java/com/google/sps/servlets/AuthServlet.java

```diff
@@ -40,66 +40,97 @@
-        }
+    Entity currentUser = queryForUser(userService);
+    if (Objects.isNull(currentUser)) {
+      setRegistrationAttributes(
+        userService,
+        urlToRedirectToAfterUserLogsOut,
+        session
+      );
+      response.sendRedirect("/profile_creation.jsp");
+    } else {
+      setUserAttributes(
+        session,
+        urlToRedirectToAfterUserLogsOut,
+        userService,
+        currentUser
+      );
+      response.sendRedirect("/user?id=" + currentUser.getKey().getId());
+    }
   }
 
-  public Entity queryForUser(UserService userService){
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      Entity currentUser = null;
-      String userEmail = userService.getCurrentUser().getEmail();
-      Filter propertyFilter = new FilterPredicate(
-                "email",
-                FilterOperator.EQUAL,
-                userEmail
-        );
-       Query q = new Query("User").setFilter(propertyFilter);
-       PreparedQuery pq = datastore.prepare(q);
-       List<Entity> userEntityList = pq.asList(FetchOptions.Builder.withLimit(1));
-       if ( userEntityList.size() >= 1){currentUser = userEntityList.get(0);}
-       return currentUser;
+  public Entity queryForUser(UserService userService) {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity currentUser = null;
+    String userEmail = userService.getCurrentUser().getEmail();
+    Filter propertyFilter = new FilterPredicate(
+      "email",
+      FilterOperator.EQUAL,
+      userEmail
+    );
+    Query q = new Query("User").setFilter(propertyFilter);
+    PreparedQuery pq = datastore.prepare(q);
+    List<Entity> userEntityList = pq.asList(FetchOptions.Builder.withLimit(1));
+    if (userEntityList.size() >= 1) {
+      currentUser = userEntityList.get(0);
+    }
+    return currentUser;
   }
-  
-  public void setLogInAttributes(int isLoggedIn, HttpSession session, String url, UserService userService) throws IOException {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = url;
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
 
-      session.setAttribute("isLoggedIn", isLoggedIn);
-      session.setAttribute("userEmail", userEmail);
-      session.setAttribute("logoutURL", logoutUrl);
+  public void setUserAttributes(
+    HttpSession session,
+    String url,
+    UserService userService,
+    Entity userEntity
+  )
+    throws IOException {
+    String userEmail = userService.getCurrentUser().getEmail();
+    String logoutUrl = userService.createLogoutURL(url);
+
+    session.setAttribute("isLoggedIn", 1);
+    session.setAttribute("userEmail", userEmail);
+    session.setAttribute("logoutURL", logoutUrl);
+    session.setAttribute("name", userEntity.getProperty("name"));
+    session.setAttribute("bio", userEntity.getProperty("bio"));
```

Comment:
I think i see why you store the name in the session, but the bio seems a bit overkill. In general, you would want to store as less information as possible in the session. Arguably, only the userId is really needed to fetch any details. This will avoid you from having discrepancy between what's stored in the database and what's being kept in the session memory. 

For now, I think it is fine, but please keep this in mind. If you have the time, a cleanup of this would be good :-)

Reply:
Thanks, I have made a note of this